src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1195,14 +1226,14 @@ private void setupRootDevice(Image image, List<BlockDeviceMapping> deviceMapping
                 }
             }
 
-            // Create a shadow of the AMI mapping (doesn't like reusing rootMapping directly)
-            BlockDeviceMapping newMapping = rootMapping.clone();
+            // pass deleteRootOnTermination to shadow of the AMI mapping
             newMapping.getEbs().setDeleteOnTermination(Boolean.TRUE);
-            //Per the documentation, "If you are creating a volume from a snapshot, you can't specify an encryption value. This is because only blank volumes can be encrypted on creation. "
-            //The root volume will always have a snapshot, so this value needs to be set to null to work correctly
-            newMapping.getEbs().setEncrypted(null);
-            deviceMappings.add(0, newMapping);
         }
+
+        newMapping.getEbs().setEncrypted(encryptEbsRootVolume);
```

Comment:
This looks like it alters behaviour, if I did not specify it should do what the default was previously. Now if i save the configuration we get either true or false. If i specify false on an encrypted ami does it become unencrypted? This is likely to cause issues with certain users configurations

Reply:
Thanks for review! 
I've updated it so it is setting default to null to make it backward compatible